clients/src/main/java/org/apache/kafka/clients/consumer/internals/DefaultBackgroundThread.java

```diff
@@ -259,17 +144,41 @@ long handlePollResult(NetworkClientDelegate.PollResult res) {
     }
 
     public boolean isRunning() {
-        return this.running;
+        return running;
     }
 
     public void wakeup() {
-        networkClientDelegate.wakeup();
+        if (networkClientDelegate != null)
+            networkClientDelegate.wakeup();
     }
 
+    @Override
     public void close() {
-        this.running = false;
-        this.wakeup();
-        Utils.closeQuietly(networkClientDelegate, "network client utils");
-        Utils.closeQuietly(metadata, "consumer metadata client");
+        closer.close(() -> {
+            log.debug("Closing the consumer background thread");
+            running = false;
+            wakeup();
+            Utils.closeQuietly(requestManagers, "Request managers client");
+            Utils.closeQuietly(networkClientDelegate, "network client utils");
+            log.debug("Closed the consumer background thread");
+            drainAndComplete();
+        }, () -> log.warn("The consumer background thread was previously closed"));
+    }
+
+
+    /**
+     * It is possible for the background thread to close before complete processing all the events in the queue. In
+     * this case, we need throw an exception to notify the user the consumer is closed.
```

Comment:
need "to" throw

Reply:
I need "to" proofread my comments more thoroughly 😄